s-string with double braces outputs extra braces

### What happened?

_No response_

### PRQL input

```elm
from foo
derive b =  s"""
    concat('{{', {a}, '}}')
"""
```


### SQL output

```SQL
SELECT
  *,
  concat('{', { a }, '}') AS b
FROM
  foo

-- Generated by PRQL compiler version:0.9.2 (https://prql-lang.org)
```


### Expected SQL output

```SQL
SELECT
  *,
  concat('{', a, '}') AS b
FROM
  foo
```


### MVCE confirmation

- [ ] Minimal example
- [ ] New issue

### Anything else?

_No response_

I tried for an hour on this but couldn't make much progress. One downside of Chumsky is that the compiler error messages are indecipherable... And GPT-4 can't help since it came out after the training cut-off.

(here's a sample, for those interested / if anyone has strategies for handling)

<details>

```
error[E0277]: the trait bound `Repeated<NoneOf<_, &str, _>>: chumsky::Parser<char, char>` is not satisfied
   --> crates/prql-parser/src/interpolation.rs:165:13
    |
165 |           .or((none_of("{"))
    |  __________--_^
    | |          |
    | |          required by a bound introduced by this call
166 | |             //none_of(just('{').not().rewind().ignore_then(just('{')))
167 | |             // (just('{').ignore_then(just("{").not()))
168 | |             // .not()
169 | |             //     just("}}").not().rewind().then(just('}')),
170 | |             .repeated()
171 | |             .at_least(1))
    | |________________________^ the trait `chumsky::Parser<char, char>` is not implemented for `Repeated<NoneOf<_, &str, _>>`
    |
    = help: the trait `chumsky::Parser<I, Vec<O>>` is implemented for `Repeated<A>`
note: required by a bound in `or`
   --> /Users/maximilian/.cargo/registry/src/github.com-1ecc6299db9ec823/chumsky-0.9.2/src/lib.rs:915:12
    |
915 |         P: Parser<I, O, Error = Self::Error>,
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `Parser::or`

error[E0599]: the method `collect` exists for struct `Or<To<Repeated<Just<char, char, _>>, Vec<char>, char>, Repeated<NoneOf<_, &str, _>>>`, but its trait bounds were not satisfied
   --> crates/prql-parser/src/interpolation.rs:172:10
    |
172 |         .collect::<String>()
    |          ^^^^^^^ method cannot be called due to unsatisfied trait bounds
    |
   ::: /Users/maximilian/.cargo/registry/src/github.com-1ecc6299db9ec823/chumsky-0.9.2/src/combinator.rs:22:1
    |
22  | pub struct Or<A, B>(pub(crate) A, pub(crate) B);
    | -------------------
    | |
    | doesn't satisfy `_: Iterator`
    | doesn't satisfy `_: Parser<_, _>`
    |
    = note: the following trait bounds were not satisfied:
            `chumsky::combinator::Or<To<Repeated<Just<char, char, _>>, Vec<char>, char>, Repeated<NoneOf<_, &str, _>>>: chumsky::Parser<_, _>`
            which is required by `&chumsky::combinator::Or<To<Repeated<Just<char, char, _>>, Vec<char>, char>, Repeated<NoneOf<_, &str, _>>>: chumsky::Parser<_, _>`
            `chumsky::combinator::Or<To<Repeated<Just<char, char, _>>, Vec<char>, char>, Repeated<NoneOf<_, &str, _>>>: Iterator`
            which is required by `&mut chumsky::combinator::Or<To<Repeated<Just<char, char, _>>, Vec<char>, char>, Repeated<NoneOf<_, &str, _>>>: Iterator`

Some errors have detailed explanations: E0277, E0599.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `prql-parser` due to 2 previous errors
done: no snapshots to review
[[Command exited with 1]]

```

</details>

---

The issue is that at https://github.com/PRQL/prql/blob/04b5174cba01bcc79837e69f0697486210d3d6dd/crates/prql-parser/src/interpolation.rs#L54, the escape parser converts `{{` to `{` as it parses the string, and doesn't reject `{`. 

I think it might be as easy as not letting the `escape` parser parse a single `{` at https://github.com/PRQL/prql/blob/04b5174cba01bcc79837e69f0697486210d3d6dd/crates/prql-parser/src/interpolation.rs#L128-L130, but this, for example, raises an error:

```diff
diff --git a/crates/prql-parser/src/interpolation.rs b/crates/prql-parser/src/interpolation.rs
index 12242184..aff8dcc1 100644
--- a/crates/prql-parser/src/interpolation.rs
+++ b/crates/prql-parser/src/interpolation.rs
@@ -126,7 +126,8 @@ fn parser(span_base: ParserSpan) -> impl Parser<char, Vec<InterpolateItem>, Erro
         });
 
     let escape = (just("{{").to('{'))
-        .chain(just("}}").not().repeated())
+        .chain((just("}}").or(just("{"))).not().repeated())
+        // .chain(just("}}").not().repeated())
         .chain(just("}}").to('}'))
         .collect::<String>()
         .map(InterpolateItem::String);
```

--- a/crates/prql-parser/src/interpolation.rs
+++ b/crates/prql-parser/src/interpolation.rs
@@ -59,13 +59,7 @@ fn parse_interpolate() {
     @r###"
     [
         String(
-            "print('",
-        ),
-        String(
-            "{hello}",
-        ),
-        String(
-            "')",
+            "print('{hello}')",
         ),
     ]
     "###);
@@ -75,13 +69,7 @@ fn parse_interpolate() {
     @r###"
     [
         String(
-            "concat('",
-        ),
-        String(
-            "{', a, '}",
-        ),
-        String(
-            "')",
+            "concat('{', a, '}')",
         ),
     ]
     "###);
@@ -93,13 +81,25 @@ fn parse_interpolate() {
     @r###"
     [
         String(
-            "concat('",
+            "concat('{', ",
         ),
+        Expr {
+            expr: Expr {
+                kind: Ident(
+                    Ident {
+                        path: [],
+                        name: "a",
+                    },
+                ),
+                span: Some(
+                    0:13-16,
+                ),
+                alias: None,
+            },
+            format: None,
+        },
         String(
-            "{', {a}, '}",
-        ),
-        String(
-            "')",
+            ", '}')",
         ),
     ]
     "###);
@@ -123,19 +123,16 @@ fn parser(span_base: ParserSpan) -> impl Parser<char, Vec<InterpolateItem>, Erro
             InterpolateItem::Expr { expr, format }
         });
 
-    let escape = (just("{{").to('{'))
-        .chain(just("}}").not().repeated())
-        .chain(just("}}").to('}'))
-        .collect::<String>()
-        .map(InterpolateItem::String);
-
-    let string = none_of('{')
+    // Convert double braces to single braces, and fail on any single braces.
+    let string = (just("{{").to('{'))
+        .or(just("}}").to('}'))
+        .or(none_of("{}"))
         .repeated()
         .at_least(1)
         .collect::<String>()
         .map(InterpolateItem::String);
 
-    escape.or(expr).or(string).repeated().then_ignore(end())
+    expr.or(string).repeated().then_ignore(end())
 }
 
 fn offset_span(base: ParserSpan, range: std::ops::Range<usize>) -> ParserSpan {

--- a/crates/prql-parser/src/interpolation.rs
+++ b/crates/prql-parser/src/interpolation.rs
@@ -74,8 +74,6 @@ fn parse_interpolate() {
     ]
     "###);
 
-    // This is incorrect — it should return an inner expression, of the `a`, and
-    // convert the `{{` and `}}` to `{` and `}` respectively.
     assert_debug_snapshot!(
         parse("concat('{{', {a}, '}}')".to_string(), span_base).unwrap(), 
     @r###"

--- a/crates/prql-parser/src/lib.rs
+++ b/crates/prql-parser/src/lib.rs
@@ -613,7 +613,12 @@ Canada
         SString:
           - String: "{?crystal_var}"
         "###);
-        parse_expr(r#"s"foo{{bar""#).unwrap_err();
+        assert_yaml_snapshot!(parse_expr(r#"s"foo{{bar""#).unwrap(), @r###"
+        ---
+        SString:
+          - String: "foo{bar"
+        "###);
+        parse_expr(r#"s"foo{{bar}""#).unwrap_err();
     }
 
     #[test]

